src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3115,7 +3082,7 @@ private TypeSymbolWithAnnotations ApplyConversion(BoundExpression operand, Conve
                     break;
             }
 
-            return TypeSymbolWithAnnotations.Create(targetType, isNullableIfReferenceType, treatUnconstrainedTypeParameterAsNullable: true);
+            return TypeSymbolWithAnnotations.Create(targetType, isNullableIfReferenceType, treatUnconstrainedTypeParameterAsNullable: isNullableIfReferenceType == true);
```

Comment:
>isNullableIfReferenceType, treatUnconstrainedTypeParameterAsNullable: isNullableIfReferenceType == true) [](start = 64, length = 104)

I think there a chance we could remove the `treatUnconstrainedTypeParameterAsNullable` parameter altogether (at least looking at NullableWalker usages).

Reply:
Limited the parameter to one public constructor, used by `NullableWalker` only.

---
In reply to: [208687280](https://github.com/dotnet/roslyn/pull/29127#discussion_r208687280) [](ancestors = 208687280)